pyqtgraph/widgets/BusyCursor.py

```diff
@@ -1,35 +1,28 @@
-from ..Qt import QtGui, QtCore, QT_LIB
+# -*- coding: utf-8 -*-
+from contextlib import contextmanager
 
-__all__ = ['BusyCursor']
+from ..Qt import QtGui, QtCore
 
-class BusyCursor(object):
-    """Class for displaying a busy mouse cursor during long operations.
+__all__ = ["BusyCursor"]
+
+
+@contextmanager
+def BusyCursor():
+    """
+    Display a busy mouse cursor during long operations.
     Usage::
 
-        with pyqtgraph.BusyCursor():
+        with BusyCursor():
             doLongOperation()
 
     May be nested. If called from a non-gui thread, then the cursor will not be affected.
     """
-    active = []
-
-    def __enter__(self):
-        app = QtCore.QCoreApplication.instance()
-        isGuiThread = (app is not None) and (QtCore.QThread.currentThread() == app.thread())
-        if isGuiThread and QtGui.QApplication.instance() is not None:
-            if QT_LIB == 'PySide':
-                # pass CursorShape rather than QCursor for PySide
-                # see https://bugreports.qt.io/browse/PYSIDE-243
-                QtGui.QApplication.setOverrideCursor(QtCore.Qt.CursorShape.WaitCursor)
-            else:
-                QtGui.QApplication.setOverrideCursor(QtGui.QCursor(QtCore.Qt.CursorShape.WaitCursor))
-            BusyCursor.active.append(self)
-            self._active = True
-        else:
-            self._active = False
-
-    def __exit__(self, *args):
-        if self._active:
-            BusyCursor.active.pop(-1)
-            if len(BusyCursor.active) == 0:
-                QtGui.QApplication.restoreOverrideCursor()
+    app = QtCore.QCoreApplication.instance()
```

Comment:
you can have `pg.mkQApp()` here, it will create an app instance if one doesn't exist and return that, or return the current instance.

Reply:
unless you don't want this to start a QApplication instance; in which case, never mind :)

Reply:
I don't think this would be an appropriate `mkQApp` situation. I don't really know the full implications of making an app here, but in either case, not making one is at least the previous behavior, so that's some sense of safety in the decision.